go.mod

```diff
@@ -57,9 +57,9 @@ require (
 	gopkg.in/guregu/null.v3 v3.4.0
 	gopkg.in/segmentio/analytics-go.v3 v3.1.0
 	gotest.tools v2.2.0+incompatible
-	k8s.io/api v0.20.14
-	k8s.io/apimachinery v0.20.14
-	k8s.io/client-go v0.20.14
+	k8s.io/api v0.28.3
```

Comment:
K8s maintains backwards compatability for the past three releases, IIRC. Supported Kube versions should probably be something we actually track. :)

Reply:
We agreed with customer support that we support non-end-of-life Kubernetes versions (so >=1.27 right now, but 1.27 is about to fall off).

Kubernetes is backwards compatible but not necessarily forward compatible. I'm worried if we bump the client past our minimum supported version, someone will rely on a feature stabilized after it. E.g. if we supported >=1.23 but the client was on 0.24 I could accidentally use indexed completions for jobs.

Reply:
So that is to say, we should cap this at 0.27 until the release after 5/24. And then we can bump to 0.28 and change our supported version in the docs.

Reply:
going into a feature branch, so pretty clear to me that we won't release this until k8s 27 is EOL

Reply:
> I was looking to see if I could get around this by having the gateways code use a newer client version than our code does. My understanding is this isn't really possible in the Go ecosystem besides if we like vendered the k8s library? I saw some replace hacks but I'm not convinced they would work on the k8s library but I could be wrong.

yeah i dont have a lot of experience with this. i wonder what pach does. maybe just "engineering discipline" (i.e. just dont write a bug lol). i'd like to somehow restrict bumping the k8s client version though. go.mod files support comments.. you could just comment not to exceed the old live k8s version and hope folks catch that in review.

meta discussion aside, the library change in this pr lgtm. and it's behind a feature flag so 👍, even better.